### PR TITLE
BAR-174: Add `--no-partial` option to `barman-cloud-wal-restore`

### LINKED ISSUE #928: Disable automatic download of partial files with `barman-cloud-wal-restore`
Currently, `barman-cloud-wal-restore` automatically downloads a `.partial` file if the requested one is absent. It would be great to introduce an option that disables this behaviour (which I think I introduced).

The option could be `--disable-partial-fallback` or `--strict-name` (`-S`)?.

Note: CloudNativePG needs this feature to provide more control over when the operator can request the download of a partial file.

### PR DESCRIPTION
**Pull request by mnencia**
Closes #928

### CODE REVIEW

**Review by gbartolini — APPROVED** [review]

**Comment by gbartolini** [comment]
Will the documentation be part of the man page (I apologise for the question, but I am not up to date anymore with Barman docs)

**Comment by barthisrael** [comment]
> Will the documentation be part of the man page (I apologise for the question, but I am not up to date anymore with Barman docs)

I would say yes -- the docs team is working with us to revamp the Barman documentation.
The information from the manual would land on some of the web pages if the current WIP layout ends up being approved  :)

**Review by barthisrael — CHANGES_REQUESTED** [review]
The PR looks good to me.
I posted a small suggestion to make the changes to the manual more complete.

I performed some test locally and it went fine:

* Archive a partial WAL file:

```console
$ barman-cloud-wal-archive -P barman-cloud --endpoint-url http://localhost:4566 s3://barman-s3-test/ pg17-postgres-default /home/vagrant/barman_home/pg17-postgres-default/streaming/00000001000000000000000F.partial
```

* Attempt to restore that WAL file with `--no-partial` flag -- no success:

```console
$ barman-cloud-wal-restore -P barman-cloud --endpoint-url http://localhost:4566 s3://barman-s3-test/ --no-partial pg17-postgres-default 00000001000000000000000F 00000001000000000000000F

$ echo $?
1

$ ls 00000001000000000000000F
ls: cannot access '00000001000000000000000F': No such file or directory
```

* Attempt to restore that WAL file without `--no-partial` flag -- success:

```console
$ barman-cloud-wal-restore -P barman-cloud --endpoint-url http://localhost:4566 s3://barman-s3-test/ pg17-postgres-default 00000001000000000000000F 00000001000000000000000F

$ echo $?
0

$ ls 00000001000000000000000F
00000001000000000000000F
```

File: doc/barman-cloud-wal-restore.1.md [hunk 1]
```diff
@@ -53,6 +53,7 @@ optional arguments:
   -t, --test            Test cloud connectivity and exit
   --cloud-provider {aws-s3,azure-blob-storage,google-cloud-storage}
                         The cloud provider to use as a storage backend
+  --no-partial          Do not download partial WAL files
```
[the remark this answers is not in the document]
Comment (mnencia) [reply doc/barman-cloud-wal-restore.1.md:57]: added

**Review by gcalacoci — APPROVED** [review]
Patch LGTM, we will merge this soon.

**Review by barthisrael — APPROVED** [review]

**Comment by martinmarques** [comment]
@gcalacoci should we squash before merging? In any other case, given the reviews this is ready to go into master

**Pull request merged**